viewer/viewer.js

```diff
@@ -5179,111 +5437,159 @@ function buildConnections(req, res, cb) {
+                }
+              }
+              process(vsrc, vdst, f, fields, connResultSets[0].resultId);
+            } // let vdst of adst
+          } // for vsrc of asrc
+          setImmediate(hitCb);
+
+        }, function (err) {
+          resultSetStatus.err = err;
+          resultSetStatus.hits = connResultSets[0].graph.hits.total;
+          if (connResultSets.length > 1) {
+            processResultSets(connResultSets.slice(1), function(baselineResultSetStatus) {
+              return cb([resultSetStatus].concat(baselineResultSetStatus));
+            });
+          } else {
+            return cb([resultSetStatus]);
+          }
+
+        }); // async.eachLimit(graph.hits.hits) / function(err)
+      } // if connResultSets[0].err) / else
+
+    } else {
+      return cb([null]);
+    } // (connResultSets.length > 0) / else
+
+  } // processResultSets
+
+  ////////////////////////////////////////////////////////////////////////////////////////////
+  // call to build the session query|queries and indices
+  buildConnectionQuery(req, fields, options, fsrc, fdst, dstipport, 1, function(connQueries) {
 
     if (Config.debug) {
-      console.log('buildConnections query', JSON.stringify(query, null, 2));
+      console.log('buildConnections.connQueries', connQueries.length, JSON.stringify(connQueries, null, 2));
     }
 
-    Db.searchPrimary(indices, 'session', query, options, function (err, graph) {
-      if (Config.debug) {
-        console.log('buildConnections result', JSON.stringify(graph, null, 2));
-      }
+    // ONE or TWO session queries will be executed, depending on if baseline is enabled:
+    //   1. for the "current" time frame, the one specified originally in req.query
+    //   2. for the "baseline" time frame immediately prior to the time frame of "1."
+    //      (only if baseline is enabled)
+    // The call to process() will ensure the resultId value is OR'ed into the .inresult
+    //   attribute of each node.
 
-      if (err || graph.error) {
-        console.log('Build Connections ERROR', err, graph.error);
-        return cb(err || graph.error);
-      }
+    // prepare and execute the Db.searchPrimary query|queries
 
-      async.eachLimit(graph.hits.hits, 10, function (hit, hitCb) {
-        let f = hit._source;
-        f = flattenFields(f);
+    if (connQueries.length > 0) {
+      dbConnectionQuerySearch(connQueries, function(connResultSets) {
 
-        let asrc = hit.fields[fsrc];
-        let adst = hit.fields[fdst];
-
-        if (asrc === undefined || adst === undefined) {
-          return setImmediate(hitCb);
+        if (Config.debug) {
+          console.log('buildConnections.connResultSets', connResultSets.length, JSON.stringify(connResultSets, null, 2));
         }
 
-        if (!Array.isArray(asrc)) {
-          asrc = [asrc];
-        }
+        // aggregate final return values for nodes and links
+        processResultSets(connResultSets, function(connResultSetStats) {
```

Comment:
in the callback to `processResultSets` we finalize everything into the arrays that will actually be returned from `buildConnections`